internal/controllers/gateway/gateway_controller_test.go

```diff
@@ -604,7 +605,7 @@ func Test_getReferenceGrantConditionReason(t *testing.T) {
 			name:             "reference granted, secret name specified",
 			gatewayNamespace: "test",
 			certRef: gatewayv1alpha2.SecretObjectReference{
-				Kind:      (*gatewayv1alpha2.Kind)(pointer.StringPtr("secret")),
+				Kind:      util.StringToKindPtr("Secret"),
```

Comment:
I noticed that we're using our own util this for this, but we already have `k8s.io/utils/pointer` and we use it in the control plane controller currently. I would suggest we just use the `pointer.String()` and other relevant functions from that package rather than maintaining our own.

Reply:
I added these utils in this PR, and they are just wrappers of `pointer.String()` and `pointer.StringPtr()`

Reply:
I don't think they'll require maintenance effort in the future, do you?

Reply:
I believe the whole point for this is to have funcs dedicated to gateway types, whereas `pointer.String()` and `pointer.StringPtr()` are only string oriented.

Reply:
My apologies. I did not do my due diligence when I made this comment, it was from the hip.